### PR TITLE
Device Schedules

### PR DESCRIPTION
**Pull request by chrismerck**
Draft "skeds" API ahead of planned "Device Schedules" feature.

### CODE REVIEW

**Comment by chrismerck** [comment]
@jdlambert OK, last commit above implemented the cases we discussed @ scrum.

(I've not determined yet how to fix the CI :/. I'm not seeing what's different versus DevicesPath->Post.)

**Comment by jdlambert** [comment]
Has this PR not been marked for review just because of the CI tests? If that's all, I can take a quick stab at it.

**Comment by chrismerck** [comment]
Go for it. Was draft because we hadn't locked it down yet. But after b7d95e6 I suppose we are good.

File: sys/paths.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      $ref: ../common/responses.yaml#/Unauthorized
+    '404':
+      $ref: ../common/responses.yaml#/NotFound
+    '500':
+      $ref: ../common/responses.yaml#/InternalServerError
+  summary: Modify time info
+  description: |
+    Set timezone, grid location, and/or time.
+
+    If `tz` is set to `null`, all skeds based on a local time (mark=midnight) will
```
Comment (chrismerck) [inline sys/paths.yaml:66]: @jdlambert

File: skeds/paths.yaml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+      $ref: ../common/responses.yaml#/BadRequest
+    '401':
+      $ref: ../common/responses.yaml#/Unauthorized
+    '500':
+      $ref: ../common/responses.yaml#/InternalServerError
+  summary: Create new Device Schedule
+  description: |
+    Creates a Device Schedule under a specific Device.
+
+    If the timezone or location settings in `sys/time` are not set,
```
Comment (chrismerck) [inline skeds/paths.yaml:51]: @jdlambert

**Comment by jdlambert** [comment]
@chrismerck 

The spec validation was failing as follows:

```
{'content': {'application/json': {'schema': {'$ref': 'schemas.yaml#/Sked', 'x-scope': ['file:///Users/jdlambert/src/olibra/api-v2/local.yaml', 'file:///Users/jdlambert/src/olibra/api-v2/local/paths.yaml', 'file:///Users/jdlambert/src/olibra/api-v2/skeds/paths.yaml#/SkedsPath', 'file:///Users/jdlambert/src/olibra/api-v2/skeds/paths.yaml#/Post']}}}} is not valid under any of the given schemas

Failed validating 'oneOf' in schema['properties']['paths']['patternProperties']['^/']['properties']['post']['properties']['requestBody']:
    {'oneOf': [{'$ref': '#/definitions/requestBody'},
               {'$ref': '#/definitions/reference'}]}

On instance['paths']['/v2/devices/{device_id}/skeds']['post']['requestBody']:
    {'content': {'application/json': {'schema': {'$ref': 'schemas.yaml#/Sked',
                                                 'x-scope': ['file:///Users/jdlambert/src/olibra/api-v2/local.yaml',
                                                             'file:///Users/jdlambert/src/olibra/api-v2/local/paths.yaml',
                                                             'file:///Users/jdlambert/src/olibra/api-v2/skeds/paths.yaml#/SkedsPath',
                                                             'file:///Users/jdlambert/src/olibra/api-v2/skeds/paths.yaml#/Post']}}}}
```

The resolution I found was to move all the `required` tags into a list in the response spec, you can see it here: https://github.com/bondhome/api-v2/pull/14/commits/c2998653793a6f75e129ee18d27d3a4838f92c32.

Just wanted to get your eyes on this fix, as I found the validation error didn't really help me find the problem, and I ended up spending too much time on this.

**Comment by jdlambert** [comment]
> Huh. Sorry it took so long to fix. Those error messages are just not helpful.
> 
> So, if changing from a `required` property to a `required` list fixed the problem, was the `required` property working but failing validation, or had I just added it there and it was never working and just causing the validation breakage?

It wasn't rendering in the docs, but without any console logging.

**Pull request merged**